src/observe.js

```diff
@@ -1685,7 +1685,8 @@
 
   var expose = global;
 
-  if (typeof exports !== 'undefined') {
+  if (typeof exports !== 'undefined' && exports + '' === '[object Object]' &&
```

Comment:
I agree. You mentioned UMD, so [here's how they do it](https://github.com/umdjs/umd/blob/master/returnExports.js#L45). It's kind of minimal (just a `typeof === 'object'` check) but it would at least fix this bug.